core/src/translate/mod.rs

```diff
@@ -165,6 +174,16 @@ fn translate_object_name(sql_object_name: &SqlObjectName) -> ObjectName {
     ObjectName(translate_idents(&sql_object_name.0))
 }
 
+fn translate_table_with_join(table: &TableWithJoins) -> Result<ObjectName> {
+    if !table.joins.is_empty() {
+        return Err(TranslateError::JoinOnUpdateNotSupported.into());
+    }
+    match &table.relation {
+        TableFactor::Table { name, .. } => Ok(ObjectName(translate_idents(&name.0))),
+        _ => Err(TranslateError::JoinOnUpdateNotSupported.into()),
```

Comment:
It looks that this could be not directly related to JOIN.
Let's add a new error with `UnsupportedTableFactor(String)` and have a corresponding test case for this.